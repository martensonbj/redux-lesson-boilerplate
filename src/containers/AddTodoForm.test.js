import React from 'react'
import { shallow } from 'enzyme'
import { AddTodoForm, mapStateToProps, mapDispatchToProps } from './AddTodoForm'
import { addTodo } from '../actions'

describe('AddTodoForm', () => {

  describe('AddTodoForm', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<AddTodoForm />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should call handleSubmit when the form is submitted', () => {
      const mockHandleSubmit = jest.fn()
      const mockTodos = []
      const mockEvent = { preventDefault: jest.fn() }
      const wrapper = shallow(<AddTodoForm handleSubmit={mockHandleSubmit} todos={mockTodos} />)

      wrapper.instance().submitForm(mockEvent)

      expect(mockHandleSubmit).toHaveBeenCalled()
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with the todos array', () => {
      // setup
      const text = 'Conquer testing'
      const id = 1
      const mockState = {
        todos: [{ text, id, completed: false }],
        filter: 'SHOW_ALL'
      }
      const expected = {
        todos: [{ text, id, completed: false }]
      }

      // execution
      const mappedProps = mapStateToProps(mockState)
      console.log(mappedProps)

      // expectation
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function from MDTP', () => {
      // setup
      const mockDispatch = jest.fn()
      const actionToDispatch = addTodo('Learn Redux', 1)

      // execution
      const mappedProps = mapDispatchToProps(mockDispatch)
      console.log(mappedProps)

      mappedProps.handleSubmit('Learn Redux', 1)

      // expectation
      expect(mockDispatch).toBeCalledWith(actionToDispatch)
    })
  })

})
