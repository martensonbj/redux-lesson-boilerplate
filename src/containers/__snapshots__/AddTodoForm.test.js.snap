// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodoForm AddTodoForm should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddTodoForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
>
        <input
                onChange={[Function]}
                placeholder="Add A Todo"
                value=""
        />
        <button>
                Add Todo
        </button>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Add A Todo"
            value=""
/>,
          <button>
            Add Todo
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Add A Todo",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Todo",
          },
          "ref": null,
          "rendered": "Add Todo",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <input
                    onChange={[Function]}
                    placeholder="Add A Todo"
                    value=""
          />
          <button>
                    Add Todo
          </button>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="Add A Todo"
              value=""
/>,
            <button>
              Add Todo
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Add A Todo",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Todo",
            },
            "ref": null,
            "rendered": "Add Todo",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
